repos/system_upgrade/el7toel8/actors/checkminimumrequirements/libraries/library.py

```diff
@@ -0,0 +1,89 @@
+            min_req_data['kernel_version'], hardware_info.kernel_version))
+
+
+def _get_memory_info(hardware_info):
+    mem_info = dict((i.split()[0].rstrip(':'), int(i.split()[1])) for i in open('/proc/meminfo').readlines())
+    total_mem_kib = mem_info['MemTotal']
+
+    hardware_info.mem_total = total_mem_kib
+
+
+def _check_memory(minimum_req_error, mem_info):
+    if (architecture.matches_architecture(architecture.ARCH_X86_64) and
+            mem_info.mem_total < min_req_data['memory'][architecture.ARCH_X86_64]):
+        minimum_req_error.append(
+            'Total Memory required: {}   found: {}'.format(min_req_data['memory'][architecture.ARCH_X86_64], mem_info))
+    if (architecture.matches_architecture(architecture.ARCH_ARM64)
+            and mem_info.mem_total < min_req_data['memory'][architecture.ARCH_ARM64]):
+        minimum_req_error.append(
+            'Total Memory required: {}    found: {}'.format(min_req_data['memory'][architecture.ARCH_ARM64], mem_info))
+    if (architecture.matches_architecture(architecture.ARCH_PPC64LE) and
+            mem_info.mem_total < min_req_data['memory'][architecture.ARCH_PPC64LE]):
+        minimum_req_error.append(
+            'Total Memory required: {}    found: {}'.format(min_req_data['memory'][architecture.ARCH_PPC64LE],
+                                                            mem_info))
+    if (architecture.matches_architecture(architecture.ARCH_S390X) and
+            mem_info.mem_total < min_req_data['memory'][architecture.ARCH_S390X]):
+        minimum_req_error.append(
+            'Total Memory required: {}    found: {}'.format(min_req_data['memory'][architecture.ARCH_S390X], mem_info))
+
+
+def check_minimum_hardware():
+    hardware_info = HardwareInfo()
+    minimum_req_error = []
+
+    _get_kernel_version(hardware_info)
+    _get_memory_info(hardware_info)
+
+    api.produce(hardware_info)
+
+    _check_kernel_version(minimum_req_error, hardware_info)
+    _check_memory(minimum_req_error, hardware_info)
+
+    if minimum_req_error:
+        title = ('Minimum hardware requirements for RHEL 8 not detected. Upgrade cannot proceed.')
+        URL = ('https://access.redhat.com/articles/rhel-limits')
+        summary = ('The following hardware requirements is not detected:'
+                   '\n{}.'.format('\n     - '.join(minimum_req_error)))
+        remediation = ('Please check your hardware and provide upgrade, see: ' + URL)
+
+        create_report([
+                          reporting.Title(title),
+                          reporting.Summary(summary),
+                          reporting.Severity(reporting.Severity.HIGH),
+                          reporting.Tags([reporting.Tags.OS_FACTS]),
+                          reporting.Flags([reporting.Flags.INHIBITOR]),
+                          reporting.Remediation(hint=remediation)
+                      ] + [reporting.RelatedResource('minimum_hardware_requirements', i) for i in minimum_req_error])
```

Comment:
There is no resource of this type no but regarding discussion with pavel, you should add resources like `disk`, `memory`, ... based on what requirements are not met.

Reply:
btw, lists of possible / existing resources, flags, tags, ... are here: https://leapp.readthedocs.io/en/latest/actor-rhel7-to-rhel8.html#reporting-tips . about adding new stuff, sync with @shaded-enmity

Reply:
Or just create a PR for adding a necessary resource type :)